[DevDocs] Various improvements and updates

djpowers opened the pull request:
## Description

Thanks to @pomdtr for the work on this extension! I recently started using it and found myself inspired to make a handful of what I considered to be quality of life improvements. I'll try to explain my rationale for the bigger changes below.

Hoping all looks good, but please let me know if you have any concerns!

### New icon
The current icon is very hard to read at small sizes, especially with the small text featured within. The icon itself is 512 x 512, but has padding that makes it look even smaller. There does not appear to be an "official" DevDocs logo, so I used https://icon.ray.so/ to try and replicate the DevDocs favicon (and webapp icon). More details in cd20a112a648558f47773ecc908872d904356db8. Certainly open to feedback here!

### Renamed command
I renamed "Search Docsets" to "Search Documentations" in order to be consistent with the phrasing used on the DevDocs site and associated materials. "Docsets" is used by the similar project Dash, so I thought that it was worth distinguishing them. I kept the original term as a keyword so those used to searching for it in Raycast would still find it.

### Updated action panel items
For the documentation commands, I added a keyboard shortcut for creating a Quicklink, and added an action to copy the slug to the clipboard (which I figured could be helpful for those who want it in their clipboard for use with the "Search Entries" command.

### Updated Quicklink default name
I included the extension name ("DevDocs") in the default name, which makes it easier to tell at a glance what the Quicklink is associated with (especially if the icon is changed). It also ensures all DevDoc Quicklinks will show up grouped together in preferences.

### Show current documentation slug in navigation
Right now there is no indication of what documentation the current entries are from. The reference to the documentation is not available from "Search Entries" and it did not seem like making an additional API just to get it, so it uses the readily available slug as a reference point.

### Refreshed screenshots
The screenshots appeared to be outdated for the current version, so I captured new ones which I believe demonstrate all of the functionality.

## Screencast

https://github.com/user-attachments/assets/83cede1b-f6ad-4524-9f43-daa0dea913b4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pomdtr commented:
Excellent improvements all around ! Would you be interested in taking ownership of the extension ?

djpowers commented:
Thanks @pomdtr, glad you approve! And I'd certainly be open to helping maintain this extension going forward. If you'd prefer to officially transfer it over that's also fine by me.

pernielsentikaer commented on extensions/devdocs/package.json:
```diff
@@ -6,13 +6,15 @@
     {
-      "name": "search-docsets",
-      "title": "Search Docsets",
+      "name": "search-documentations",
```
Can we keep the old name here? It's the unique ID of the command where ranking, aliases and hotkeys are saved

pernielsentikaer replied:
@djpowers 🙂

djpowers replied:
Hello! Busy day today, have since reviewed. So changing the `name` property would lose the link to existing aliases/hotkeys/rankings? I'd prefer to have it match the new title, but if there's otherwise no migration path to avoid a breaking change then I'm fine keeping the original name.

Will make that change along with some other cleanup.

pernielsentikaer commented:
@pomdtr do you want to be moved to `pastContributors` or just `contributors` if we transfer it to @djpowers?

pomdtr commented:
You can move me to past contributors, I barely use the extension

djpowers commented:
Should be ready for another look! I reverted the change to the `name` value for the "search documentations" command, made some changes to ensure the Quicklink creation still worked with the ownership change, added some metadata, and tweaked the icons/screenshots a little.

Thanks again to @pomdtr for the initial work and laying down a solid foundation to build upon!

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
